### ISSUE TITLE
/* SQL Error (1034): Incorrect key file for table 'cache_object_count'; try to repair it */

### ISSUE THREAD

**Issue by cnic023** [ampache/ampache#2558]
Hi. I got this error then dropped the ampache DB, created one with the same name and installed my latest backup. Everything fine until I tried to create a playlist. No playlist could be saved and the DB was corrupted again with the same error message. I'm using Ubuntu server 20.4 and the latest Ampache.
Is it possible to get a clean mysql backup of just the table structures? I'm wondering if something got corrupted before my last backup?

**Comment by lachlan-00** [comment]
you can safely drop that table and recreate it

```
DROP TABLE IF EXISTS `cache_object_count`;
CREATE TABLE IF NOT EXISTS `cache_object_count` (
  `object_id` int(11) UNSIGNED NOT NULL,
  `object_type` enum('album','artist','song','playlist','genre','catalog','live_stream','video','podcast_episode') CHARACTER SET utf8 COLLATE utf8_general_ci NOT NULL,
  `count` int(11) UNSIGNED NOT NULL DEFAULT '0',
  `threshold` int(11) UNSIGNED NOT NULL DEFAULT '0',
  `count_type` varchar(16) CHARACTER SET utf8 COLLATE utf8_general_ci NOT NULL,
  PRIMARY KEY (`object_id`,`object_type`,`threshold`,`count_type`)
) ENGINE=MyISAM DEFAULT CHARSET=utf8 COLLATE=utf8_unicode_ci;
```

**Comment by cnic023** [comment]
Many thanks for your help.
I now get for multiple tables:
/* SQL Error (1034): Incorrect key file for table 'cache_object_count_run'; try to repair it */
Any suggestions?
Ampache is great!!!
Chris

**Comment by wagnered** [comment]
Here is a suggestion from the [Planet MySQL](https://planet.mysql.com/entry/?id=32629) archives. It points to info for fixing tables. It also describes how to check for disk space on Linux or Windows.  It seems to apply to your situation.

**Comment by cnic023** [comment]
Thanks Ernest. I saw this link when I first got the error message but my disk space seemed ok. However, in case I missed something details are attached.
I think my problem has something to do with creating a playlist. As I only have a few albums in my collection I’m wondering if I should start again to make sure I have a clean database and then see if the problem reoccurs. I have a backup created with mysqldump but that perhaps is corrupt (although all tables were ok before I tried to make a playlist). I really want to be sure I have a clean backup of the correct table structures.
Chris

**Comment by wagnered** [comment]
Sorry but nothing was attached.

From what I could find out, corrupt tables would have prevented  a dump of the tables.  So the dump should be good.

I would like to see any error details.

MySQL might not have the most robust error reporting with queries involving multiple joined tables.  I don't know if the query does involve multiple joins, but . MySQL might be reporting the wrong table.

There is a primer for fixing MyISAM
tables and recovering InnoDB tables at [several nines](https://severalnines.com/database-blog/my-mysql-database-corrupted-what-do-i-do-now).

On the physical side, how old is your drive? in case there is a bad spot on the disk moving the MySQL table location might fix the problem, depending on the file system.
Ok

**Comment by cnic023** [comment]
Hi
I have attached the disk information. The server is on a Lenovo B50 30 which is about 6 years old.
I have now restored the database and it shows no errors. However, I have not yet tried to create a playlist. My backup file is also enclosed.
The present database after backup does not show a table cache_object_count or cache_object_count_run. The table object_count shows no errors. I don’t know how relevant this is.
Cheers
Chris

**Comment by cnic023** [comment]
I have run a mysqlcheck and the results are in the attached file.
I have attempted repair using myisamchk and this was not successful.

**Comment by wagnered** [comment]
I don't know if my isp (comcast) is blocking your attachments, but I haven't received any.

After logging into GitHub, could you create a [GitHub gist](https://gist.github.com/) and add the results of your tests?  I should be able to see and download any public gist you create.

**Comment by cnic023** [comment]
I have made a gist called ‘Incorrect key file for table in Ampache SQL DB’

**Comment by cnic023** [comment]
I ran a disk check that revealed no errors

**Comment by wagnered** [comment]
All my web searches indicate disk corruption or insufficient memory/disk/ partition space..

**Comment by cnic023** [comment]
I accept your conclusion. I will get a new server with RAID capabilities and reinstall. Do you have any advice on suitable hardware for home/small office use? I appreciate that this may be outside your brief. Anyway thanks again for you help

**Comment by wagnered** [comment]
> I accept your conclusion. I will get a new server with RAID capabilities and reinstall.

Before you do, you might want to try [Relocating Data Directory Contents](http://etutorials.org/SQL/MySQL/Part+III+MySQL+Administration/Chapter+10.+The+MySQL+Data+Directory/Relocating+Data+Directory+Contents/) to a different partition or drive?  That reference is a tutorial and should apply to Maria db.  

If it resolves your immediate problem, it might give you peace of mind to know the drive is going "belly up".  6 years is quite a long time for a drive, especially if it is on a server and active  24/7.

I have used RAID configuration in the past so I know a little.   I started off with a little one TB 2 drive hardware RAID1. Then I used the drives from that to create a software RAID1 on a freebsd system.  I suggest you read up on both types of RAID systems.

If you go software, I suggest a *BSD system.  Their emphasis is on disk management and a GUI is not automatically installed, although an X11 system can be installed.
